docs/content/how-to/baremetal-none.md

```diff
@@ -0,0 +1,443 @@
+
+~~~sh
+oc apply -f ./hypershift-install.yaml
+~~~
+
+## Deploy a hosted cluster
+
+There are two main CRDs to describe a hosted cluster:
+
+* [`HostedCluster`](https://hypershift-docs.netlify.app/reference/api/#hypershift.openshift.io/v1alpha1.HostedCluster) defines the control plane hosted in the management OpenShift
+
+* [`NodePool`](https://hypershift-docs.netlify.app/reference/api/#hypershift.openshift.io/v1alpha1.NodePool) defines the nodes that will be created/attached to a hosted cluster
+
+The `hostedcluster.spec.platform` specifies the underlying infrastructure provider for the cluster and is used to configure platform specific behavior, so depending on the environment it is required to configure it properly.
+
+In this repo we will cover the 'none' provider.
+
+### Requisites
+
+* Proper DNS entries for the workers (if the worker uses `localhost` it won't work)
+* DNS entry to point `api.${cluster}.${domain}` to each of the nodes where the hostedcluster will be running. This is because the hosted cluster API is exposed as a `nodeport`. For example:
+
+~~~sh
+api.hosted0.example.com.  IN A  10.19.138.32
+api.hosted0.example.com.  IN A  10.19.138.33
+api.hosted0.example.com.  IN A  10.19.138.37
+~~~
+
+* DNS entry to point `*.apps.${cluster}.${domain}` to a load balancer deployed to redirect incoming traffic to the ingresses pod [the OpenShift documentation](https://docs.openshift.com/container-platform/4.9/installing/installing_platform_agnostic/installing-platform-agnostic.html#installation-load-balancing-user-infra-example_installing-platform-agnostic) provides some instructions about this)
+> **NOTE**: This is not strictly required to deploy a sample cluster but to access the exposed routes there. Also, it can be simply an A record pointing to a worker IP where the ingress pods are running and enabling the `hostedcluster.spec.infrastructureAvailabilityPolicy: SingleReplica` configuration parameter.
+
+* Pull-secret (available at cloud.redhat.com)
+* ssh public key already available (it can be created as `ssh-keygen -t rsa -f /tmp/sshkey -q -N ""`)
+* Any httpd server available to host a ignition file (text) and a modified RHCOS iso
+
+### Procedure
+
+* Create a file containing all the variables depending on the enviroment:
+
+~~~sh
+cat <<'EOF' > ./myvars
+export CLUSTERS_NAMESPACE="clusters"
+export HOSTED="hosted0"
+export HOSTED_CLUSTER_NS="clusters-${HOSTED}"
+export PULL_SECRET_NAME="${HOSTED}-pull-secret"
+export MACHINE_CIDR="10.19.138.0/24"
+export OCP_RELEASE_VERSION="4.9.17"
+export OCP_ARCH="x86_64"
+export BASEDOMAIN="example.com"
+
+export PULL_SECRET_CONTENT=$(cat ~/clusterconfigs/pull-secret.txt)
+export SSH_PUB=$(cat ~/.ssh/id_rsa.pub)
+EOF
+source ./myvars
+~~~
+
+* Create a namespace to host the HostedCluster and secrets
+
+~~~sh
+envsubst <<"EOF" | oc apply -f -
```

Comment:
Is there a reason for templating all the yaml files vs using `hypershift create cluster` ?

I know it's basically the same thing, but for new users running the CLI may be a bit simpler and easier to follow in the docs (and we could mention the render option if the CLI generated config needs to be customized)

Reply:
I guess for the sake of learning is better to showcase the CRDs, their specs, etc. instead of blindly relaying on the binary command, also for automation purposes (or maybe to integrate it with a gitops flow) I think it is better to have the yamls instead of running the binary. But that's just my 2 cents :)

Reply:
Fair enough, I was thinking we'd want to make the howto docs as simple as possible for new users - and it seems weird that we describe how to build/install the CLI then don't use it, I can see your point though so happy to go with the consensus :)

Reply:
In fact the cli is built just for the purpose of bootstraping the operator (`hypershift install --render`). Maybe we can expose both ways using something like an spoiler tag (not sure if netlify supports it though)